### PR TITLE
Allow sharing of investigations

### PR DESCRIPTION
**Pull request by tillprochaska**
This enables sharing of investigations with individual users by providing their full email address. Main changes are in 7cced50eaa0ff09eca17dba5c977c9b6814ac6c7.

If you want to manually test cases including OAuth user groups, you can set up Keycloak in your development environment like this: https://docs.aleph.occrp.org/developers/how-to/development/identity-provider/

### CODE REVIEW

File: aleph/tests/test_permissions_api.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
+from aleph.tests.util import TestCase
+from aleph.logic.roles import create_group
+
+
+class PermissionsApiTestCase(TestCase):
```
Comment (stchris) [inline aleph/tests/test_permissions_api.py:5]: Thanks for this! 🙏

File: aleph/views/roles_api.py [hunk 2]
```diff
@@ -51,12 +54,26 @@ def suggest():
       - Role
     """
     require(request.authz.logged_in)
-    return jsonify(
-        {
-            "results": [],
-            "total": 0,
-        }
+    parser = QueryParser(request.args, request.authz, limit=10)
+    if parser.prefix is None or len(parser.prefix) < 3:
```
Comment (stchris) [inline aleph/views/roles_api.py:58]: I guess we could raise minimum length to 5 or 6 here, but it's a minor thing.

**Review by stchris — APPROVED** [review]
LGTM, thanks for this! Left some minor remarks which are fine to be considered later (or not at all)

File: aleph/views/roles_api.py (hunk 2 above)
Comment (tillprochaska) [reply aleph/views/roles_api.py]: Done!

**Pull request merged**